lib/codecov.js

```diff
@@ -159,10 +161,26 @@ var upload = function(args, on_success, on_failure){
 ' \\_____\\___/ \\__,_|\\___|\\___\\___/ \\_/  \n' +
 '                                ' + version);
 
+  var token = (args.options.token || process.env.codecov_token || process.env.CODECOV_TOKEN);
+
+  if (token) {
+    query.token = token;
+  }
+
   query.yaml = ['codecov.yml', '.codecov.yml'].reduce(function (result, file) {
     return result || (fs.existsSync(path.resolve(process.cwd(), file)) ? file : undefined)
   }, undefined)
 
+  // TODO: we should set other values here too based off yaml file
+  // if a yaml file was provided, then parse it and look for the token (gh #23)
+  if (query.yaml) {
+    var yaml = YAML.load(path.resolve(process.cwd(), query.yaml));
+    if (yaml.codecov && yaml.codecov.token)
+      query.token = yaml.codecov.token;
+    else
+      throw new Error('yaml file detected but `codecov` block missing');
```

Comment:
That's intended?  If you don't have both then it should fail.  Right now I
only have support for YAML token, which I'm assuming people would use?

Can you submit a PR instead that adds support for populating all options
based off YAML options?

On Mon, Jul 25, 2016 at 10:52 AM, Ed Moore <notifications@github.com> wrote:

> In lib/codecov.js
> <https://github.com/codecov/codecov-node/pull/25#discussion_r72077988>:
>
> >    query.yaml = ['codecov.yml', '.codecov.yml'].reduce(function (result, file) {
> >      return result || (fs.existsSync(path.resolve(process.cwd(), file)) ? file : undefined)
> >    }, undefined)
> >
> > +  // TODO: we should set other values here too based off yaml file
> > +  // if a yaml file was provided, then parse it and look for the token (gh #23)
> > +  if (query.yaml) {
> > +    var yaml = YAML.load(path.resolve(process.cwd(), query.yaml));
> > +    if (yaml.codecov && yaml.codecov.token)
> > +      query.token = yaml.codecov.token;
> > +    else
> > +      throw new Error('yaml file detected but `codecov` block missing');
>
> Ok. Say you have the yaml file that has the codecov block. But inside that
> block the token key doesn't exist. Then the if statement would be falsy,
> therefore it will hit the else statement.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/codecov/codecov-node/pull/25/files/0aeab66b6d3f2f53b1e8b0282ae56271a2664f82#r72077988>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAf7hY20JXBOz5Mq7Lm8Qr80nA_Qa27hks5qZM2egaJpZM4JUCSL>
> .
>



-- 
@niftylettuce